packages/spy/src/index.ts

```diff
@@ -562,16 +585,31 @@ function enhanceSpy<T extends Procedure>(
   }
 
   function mockCall(this: unknown, ...args: any) {
-    instances.push(this)
-    contexts.push(this)
+    if (!new.target) {
+      instances.push(this)
+      contexts.push(this)
+    }
     invocations.push(++callOrder)
     const impl = implementationChangedTemporarily
       ? implementation!
       : onceImplementations.shift()
         || implementation
         || state.getOriginal()
-        || (() => {})
-    return impl.apply(this, args)
+        || function () {}
+    if (new.target) {
+      try {
+        const result = Reflect.construct(impl, args, new.target)
+        instances.push(result)
+        contexts.push(result)
+        return result
+      }
+      catch (e) {
+        instances.push(undefined)
+        contexts.push(undefined)
+        throw e
+      }
```

Comment:
Can we detect the incorrect usage here? If yes, the error message could point to documentation too. 

```ts
if (e instanceof TypeError && e.message.endsWith('is not a constructor')) {
  e.message = `vi.spyOn() for class did not use 'function' or 'class', see https://vitest.dev/api/vi.html#vi-spyon for examples.\n\n${e.message}`
}
```